src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -5046,6 +5046,8 @@ def set_up_agentpool_profile(self, mc: ManagedCluster) -> ManagedCluster:
         self._ensure_mc(mc)
 
         agentpool_profile = self.agentpool_decorator.construct_agentpool_profile_default()
+        # set up vm set type and also can be customized from AKS-Preview cli extension
+        agentpool_profile = self.agentpool_decorator.set_up_agentpool_type(agentpool_profile)
```

Comment:
Should put this setter inside the constructer if newly created agentpool could choose vm type by themselves.

Reply:
I have another change at aks-preview to set the customized agent pool type, and it works with the change here at azure-cli.

Reply:
https://github.com/Azure/azure-cli-extensions/pull/6591. But this change depends on this refactoring, otherwise the test won't pass